best-practices/tech_lead.md

```diff
@@ -0,0 +1,367 @@
+- knock down roadblocks, or help find another way around
+- make sure pull requests (PRs) are reviewed promptly, and with an eye to quality and coding goals (below)
+- ensure team awareness of current priorities, which do shift
+- keep work on track
+- try to anticipate gotchas and address them
+- make a technical decision to avoid deadlocks when the team disagrees
+
+The TL may not be the expert in the tech
+- use the team ... tap into expert knowledge wherever it resides
+
+The TL may not be able to code much
+- more devs --> more TL time needed coordinating
+- never take blocking tickets
+  - it's more important for you to be available for pairing, consulting with Product Owner (PO), ticket writing, board grooming, etc.
+  - TL meeting and communication load is higher, so either the blocking ticket will languish while you are doing TL duties or the communication and pairing will suffer while you do blocking work.  Neither is a good outcome.
+- take tiny tickets so they won't languish half done
+- consider taking unfun tickets so other devs can be happier
+  - e.g. review who has access to endpoints: this can be done in small pieces, and no one will be excited to do it
+
+_“What am I uniquely capable of doing?”_ - do these things; leave the rest on the board for others.  As TL, you are in a key position for information flow. Honor that via ticket writing, board grooming, pairing, etc.
+
+It helps to know who you are, what your skills and talents are, and to ask for help when you need it.
+
+## Project Management Has to be Done
+
+... and team members are happier if it is done well.
+
+### It's About Process
+
+>"meet with stakeholders outside of the team keeping a good information flow, or to remove blockers." https://www.patkua.com/blog/the-definition-of-a-tech-lead/
+
+There needs to be a transparent process that:
+- ___documents the work cycle and sprint goals___ for easy lookup (e.g. epics, milestones)
+- ___enables devs to easily choose the highest priority work for the sprint and work cycle___ (e.g. ready column)
+- ___keeps all devs actively engaged in working towards sprint and work cycle goals___ (e.g. population of ready column, easy way for devs working on a ticket to surface problems and get resolution quickly)
+- ___makes it easy for the TL, the PO and management to see which work is in progress and how it relates to the goals___ (e.g. the board as a whole)
+
+If you're TL, and the work-cycle PO is mostly unavailable, and there's no appointed scrum master ... project management still needs to be done.  You don't have to do everything yourself ... but you are best placed to ensure it all gets done.  This is because the PO is often not on the infrastructure team, or even in DLSS, and thus is less informed and perhaps less invested in the quality of the infrateam process. But the PO and other team members may be willing to step up if you ask for help.
```

Comment:
What if there is a scrum master in the work cycle? How does that change things below?